### ISSUE TITLE
structure padding

### ISSUE THREAD

**Issue by caughtbyatoe** [ispc/ispc#959]
the c++ structures generated via the '-h' option to the 1.8.1 ispc compiler do not include the necessary padding to ensure correct c++/ispc interoperability Consider:

test.ispc
-------------
struct MyInt
{
    int i;
};

struct MyStruct
{
    uniform int type;
    MyInt       i;
};

export void
ispc_test(varying MyStruct * uniform result)
{
    print("sizeof(MyStruct) %\n", sizeof(MyStruct));
}

test.cc
----------
#include <iostream>
#include "test_ispc.h"

int main() {
    std::cout << "sizeof(ispc::v8_varying_MyStruct) " << sizeof(ispc::v8_varying_MyStruct) << std::endl;
    ispc::ispc_test(NULL);
}

When compiled as:

ispc test.ispc --target=avx1-i32x8 -o test_ispc.o -h test_ispc.h
c++ test.cc test_ispc.o

The produced output is:

sizeof(ispc::v8_varying_MyStruct) 36
sizeof(MyStruct) 64

In other words, in order to make MyStruct usable in both c++ and ispc, we need to manually add padding between 'type' and 'i' in MyStruct.

**Comment by dbabokin** [comment]
Fixed by #968
